code/modules/mob/living/say.dm

```diff
@@ -88,11 +88,20 @@ GLOBAL_LIST_INIT(department_radio_keys, list(
 
 	var/static/list/one_character_prefix = list(MODE_HEADSET = TRUE, MODE_ROBOT = TRUE, MODE_WHISPER = TRUE)
 
+	var/ic_blocked = FALSE
+	if(client && !forced && config.ic_filter_regex && findtext(message, config.ic_filter_regex))
+		//The filter doesn't act on the sanitized message, but the raw message.
+		ic_blocked = TRUE
+
 	if(sanitize)
 		message = trim(copytext(sanitize(message), 1, MAX_MESSAGE_LEN))
 	if(!message || message == "")
 		return
 
+	if(ic_blocked)

```

Comment:
Could you log this to the admins too? It's useful to know who's attempting to use prohibited language, also lets us keep an eye on them in case they try to work around the system.

Reply:
WIll do